tests/index.js

```diff
@@ -0,0 +1,199 @@
+'use strict';
+
+import sinon from 'sinon';
+import test from 'tape';
+import {
+  Status,
+  callAPI,
+  shouldCallAPI
+} from '../build';
+
+/*
+ * SUCCESSFUL CALL
+ */
+test('Checking a successful API call', (t) => {
+  // Mock payload
+  const payload = {
+    foo: 'bar'
+  };
+
+  // Mock api call as promise with external control
+  var resolve, reject;
+  const mockAPICall = new Promise((resolver, rejecter) => {
```

Comment:
Would be nice to have some functions that generate these things, rather than each test having to repeat these steps

```js
test('Checking a successful API call', (t) => {
  var mockDeferred = createMockApiCallPromise();
  var mockAction = createMockActionWithPromise(deferred.promise);
  var mockStore = createMockStore();
  var mockNext = sinon.spy();

  var middleware = callAPI(mockStore)(mockNext);
  middleware(mockAction);

  expect(mockStore.dispatch).to.have.been.calledWith(..);

  deferred.resolve('foobar').then(() => {
    ...
  })
});
```